Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2236,6 +2236,27 @@ extension BrowserViewController: WKNavigationDelegate {
             return
         }
 
+        // Handles custom mailto URL schemes.
+        if url.scheme == "mailto" {
+            if let mailToMetadata = url.mailToMetadata() {
+                let mailScheme = self.profile.prefs.stringForKey("MailToOption") ?? ""
+                var mailURL = url
+                switch (mailScheme) {
+                case "readdle-spark://":
+                    if (!UIApplication.sharedApplication().canOpenURL(ReaddleSparkIntegration.newEmailURLFromMetadata(mailToMetadata))) {
+                        profile.prefs.setString("mailto", forKey: "MailToOption")
+                    } else {
+                        mailURL = ReaddleSparkIntegration.newEmailURLFromMetadata(mailToMetadata)
```

Comment:
There's a couple of things here that are a bit wonky.

1. You're creating the `ReaddleSparkIntegration.newEmailURLFromMetadata(mailToMetadata)` twice: once in the `canOpenURL` call and again here. You can just call it once them run the check on it.

2. Having the switch statement here on the scheme seems like duplicated effort since we already have the scheme in a MailSchemes.plist file. I would leverage the fact you already have the schemes listed as strings available to index through. 

3. I would also try to move as much of this code out of the BVC. Maybe put it in a basic class that we instantiate and add to the BVC.

With regards to (2 + 3), you can encapsulate the loading of the .plist file and generate an mapping of all the mailto schemes to providers when it initializes and when you want to open the actual mailto link, index into the dictionary to see if it exists instead of running through a switch.

```swift
class MailtoLinkHandler {

let mailSchemeProviders: [String: MailProvider]

func launchMailClientForScheme(scheme: String, url: NSURL) {
    guard let provider = mailSchemeProviders[scheme] {
        UIApplication.sharedApplication().openURL(url)
        return
    }

    let mailURL = provider.mailtoFromURL(url)
    if UIApplication.sharedApplication().canOpenURL(mailURL) {
        UIApplication.sharedApplication().openURL(mailURL)
    } else {
        UIApplication.sharedApplication().openURL(url)
    }
}
}
```